main/src/ca/uwaterloo/flix/language/ast/Scheme.scala

```diff
@@ -113,14 +131,19 @@ object Scheme {
     //
 
     // Instantiate every variable in `sc1` as flexible and make every free variable rigid.
-    val tpe1 = instantiate(sc1, InstantiateMode.Mixed)
+    val (tconstrs1, tpe1) = instantiate(sc1, InstantiateMode.Mixed)
 
     // Instantiate every variable in `sc2` as rigid and make every free variable rigid.
-    val tpe2 = instantiate(sc2, InstantiateMode.Rigid)
+    val (tconstrs2, tpe2) = instantiate(sc2, InstantiateMode.Rigid)
 
     // Attempt to unify the two instantiated types.
     Unification.unifyTypes(tpe1, tpe2) match {
-      case Result.Ok(_) => true
+      case Result.Ok(subst) =>
+        val newTconstrs1 = tconstrs1.map(subst.apply)
+        val newTconstrs2 = tconstrs2.map(subst.apply)
+        // type constraints on tvars in `sc1` must be a subset of those in `sc2`
+        // we ignore type constraints on non-vars for the purposes of the leq check
+        newTconstrs1.filter(_.arg.isInstanceOf[Type.Var]).forall(newTconstrs2.contains)
```

Comment:
Subset in what sense? Like syntactically?

Reply:
I was thinking that it would be nice with this PR if we can get to the point where you declare a TC, e.g. Show.

Then if you write:

```
def foo(x: a): String = show x
```

You should get an error, because the type signature must be:
```
def foo[a: Show](x: a): String = show x
```

(or whatever syntax).

How much work do you think it requires to get to that point?

Reply:
> Subset in what sense? Like syntactically?

In order for scheme A ≤ B: B needs to contain all the constraints in A. This is equivalent to all inferred constraints in a def being included in the declared constraints.

> How much work do you think it requires to get to that point?

I think it's pretty close. This modification to Scheme sets us up for throwing an error in this case. Without an implementation of instances, we can't really have a positive test case, but a negative case I think should work.

Reply:
I am ok with the current impl, but I am not sure it is totally correct. For example, when it comes to effects we have that `Pure = x or (not x)` which is not a simple syntactic check. But let us defer that till later.

Reply:
You're right, this will need to be de-naïved to use reduction/unification rules.